my_scripts.rb

```diff
@@ -0,0 +1,94 @@
+module Enumerable
+  def my_each(var)
+    if block_given?
+      for i in 0...var.length
+        yield(var[i])
+      end
+      var
+    else 'Error: no block given'
+    end
+  end
+
+  def my_each_with_index(var)
+    if block_given?
+      for i in 0...var.length
+        yield(var[i], i)
+      end
+      var
+    else 'Error: no block given'
+    end
+  end

```

Comment:
**1-** my_each_with_index should returns an Enumerator if no block is given:
`p [1, 2, 3].my_each_with_index` returns `#<Enumerator>`
<img width="457" alt="2" src="https://user-images.githubusercontent.com/58504655/103807613-78ec9400-5078-11eb-9679-4420ee943788.PNG">

**2-**